gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -322,8 +339,73 @@ gst_tensor_src_iio_init (GstTensorSrcIIO * self)
   self->silent = DEFAULT_PROP_SILENT;
   self->buffer_capacity = DEFAULT_BUFFER_CAPACITY;
   self->sampling_frequency = DEFAULT_FREQUENCY;
+  self->merge_channels_data = DEFAULT_MERGE_CHANNELS;
+  self->is_tensor = FALSE;
+  self->config.tensors = NULL;
+
+  /**
+   * format of the source since IIO device as a source is live and operates
+   * at a fixed frequency, GST_FORMAT_TIME is used
+   */
+  gst_base_src_set_format (GST_BASE_SRC (self), GST_FORMAT_TIME);
+  /** set the source to be live */
+  gst_base_src_set_live (GST_BASE_SRC (self), TRUE);
+}
+
+/**
+ * @brief merge multiple other/tensor
+ * @note they should have matching type and shape to form 1 other/tensors
+ * @note extra dimension should be available for other/tensors
+ * @note order of merge is stable
+ * @param[in/out] info Tensor info to be merged
+ * @param[in] size Info array size
+ * @returns >=0 number of valid entries in the info after merge
+ *         -1 failed due to missing extra dimension/mismatch shape/type
+ * @todo merge tensors (in stable order) into buckets based on type
+ */
+static gint
+gst_tensor_src_merge_tensor_by_type (GstTensorInfo * info, guint size)
+{
+  gint info_idx, base_idx, dim_idx;
+  gboolean mismatch = FALSE, dim_avail = TRUE;
+
+  /** base error control check */
+  g_return_val_if_fail (size > 0, 0);
+  base_idx = 0;
+
+  /** verify extra dimension */
+  if (info[base_idx].dimension[NNS_TENSOR_RANK_LIMIT - 1] != 1) {
+    dim_avail = FALSE;
+  }
+
+  /** verify that all the types and shapes match */
+  for (info_idx = 0; info_idx < size; info_idx++) {
+    if (!gst_tensor_info_is_equal (info + base_idx, info + info_idx)) {
+      mismatch = TRUE;
+      break;
+    }
+  }
+
+  /** return original if cant be merged and size within limits */
+  if (mismatch || dim_avail) {
```

Comment:
plz check the condition dim_avail

Reply:
I missed it in the update.
Thanks for pointing out.
Done.